mlflow/pyfunc/__init__.py

```diff
@@ -690,7 +690,7 @@ def predict(*args):
             result = result.applymap(str)
 
         if type(result_type) == ArrayType:
-            return pandas.Series([row[1].values for row in result.iterrows()])
+            return pandas.Series(result.to_numpy().tolist())
```

Comment:
@jgc128 Can we remove `.tolist()`?

Reply:
We can't - if we remove `.tolist()`, `pd.Series` would throw an exception that the input data must be 1-dimensional. 
The goal here is to create a series where each element is not a single number but an array. It's not idea but that's a limitation of Spark's pandas_udf API, as it should return a series and not a dataframe.

Reply:
Makes sense. We create a pandas sereis like below:

```python
pandas.Series([[1,], [2,]])
```

Reply:
Confriemd the following code throws:

```
pandas.Series(np.array([[1,], [2,], ...]))
```

Reply:
Yeah, if it's an array, pd.Series would check for dimensions